Let's explore an interesting example based on the dataset. Let's explore an interesting example based on the dataset. First, upload the dataset named 'data.csv'.

In [1]:
from google.colab import files


uploaded = files.upload()


Saving data.csv to data.csv


Here, we are importing all the necessary modules.

In [2]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd



Next, we have to read the dataset using the Pandas data frame as follows:

In [3]:
df = pd.read_csv("data.csv")

Let's check the shape of the data.

In [13]:
df.shape

(36, 5)

Now, observe the data.

In [12]:
df.head()

Car       Model  Volume  Weight  CO2
0      Toyoty        Aygo    1000     790   99
1  Mitsubishi  Space Star    1200    1160   95
2       Skoda      Citigo    1000     929   95
3        Fiat         500     900     865   90
4        Mini      Cooper    1500    1140  105

Now we create the lists of the independent values called X and the dependent values called y respectively.

In [4]:
X = df[['Weight', 'Volume']]
y = df['CO2']

We shall import the sklearn module as there are a few methods from this module will be useful.

In [5]:
from sklearn import linear_model

The sklearn module will provide us with the LinearRegression() method for creating a linear regression object which uses fit() method by taking the independent and dependent values as parameters by filling the regression object with data that describes the relationship:

In [6]:
regr = linear_model.LinearRegression()
regr.fit(X, y)

LinearRegression()

Finally, we have a regression object that is ready to predict CO2 values based on a car's weight and volume:

In [7]:
#predict the CO2 emission of a car where the weight is 2300kg, and the volume is 1300cm3:
predictedCO2 = regr.predict([[2300, 1300]])
print(predictedCO2)

[107.2087328]


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


Here, we predicted that a car with 1.3-liter engine, and a weight of 2300 kg, will release approximately 107.209 grams of CO2 per kilometer it drives.

**Note:**
Remember, 1 liter (L) = 1,000 cubic centimeters (cm³)

A 1.3-liter car engine has the capacity for around 1,300cc (1.3 litres) of fuel and air which is burnt with every revolution the engine turns.

In [8]:
print(regr.coef_)

[0.00755095 0.00780526]


The result array represents the coefficient values of weight and volume.

Weight Coefficient: 0.00755095

Volume Coefficient: 0.00780526

Now, we shall execute the same example by changing the weight from 2300 to 3300:

In [9]:
import pandas
from sklearn import linear_model

df = pandas.read_csv("data.csv")

X = df[['Weight', 'Volume']]
y = df['CO2']

regr = linear_model.LinearRegression()
regr.fit(X, y)

predictedCO2 = regr.predict([[3300, 1300]])

print(predictedCO2)

[114.75968007]


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:465: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


Finally, we predicted that a car with 1.3-liter engine, and a weight of 3300 kg, will release approximately 114.76 grams of CO2 for per kilometer it drives.